pkg/plugin/run.go

```diff
@@ -71,7 +72,39 @@ func (e *AutoscaleEnforcer) runPermitHandler() {
 	})
 	server := http.Server{Addr: "0.0.0.0:10299", Handler: mux}
 	klog.Info("[autoscale-enforcer] Starting resource request server")
-	klog.Fatalf("[autoscale-enforcer] Resource request server failed: %s", server.ListenAndServe())
+
+	// in general this isn't the right way to do this: we should
+	// have a group of functions which return service objects that
+	// have blocking close routines, and then start them at a
+	// higher level in the process, and then when we catch a
+	// shutdown signal, you call the shutdown methods (preferably
+	// in parallel) and wait for them to shut down.
+	go func() {
+		// this runs until something cancels it
+		if err := server.ListenAndServe(); err != nil && errors.Is(err, http.ErrServerClosed) {
+			// this fatal will take down the entire process,
+			klog.Fatalf("[autoscale-enforcer] Resource request server failed: %s", err)
+		}
+	}()
+
+	go func() {
```

Comment:
`runPermitHandler` is already launched in a new goroutine. I's say it's _probably_ (?) better to leave the contents of this function inside its caller (i.e. don't launch a new goroutine here).

Reply:
I think it doesn't really matter, but I turned it around.